### ISSUE TITLE
Custom Handle Selector Does Not Work When for Directive with TemplateUrl

### ISSUE THREAD

**Issue by mikehaas763** [ManifestWebDesign/angular-gridster#277]
I'm having a problem when I have a custom directive inside of the gridster-item directive. My custom directive uses templateUrl to automatically fetch the template async. With templateUrl there is a race condition; most of the time (for me) the custom handler does not yet exist when the `querySelector` code is ran so it doesn't get used. When I debug and look at the html contents of `$el[0]` in GridsterDraggable I see something like this:
```html
<div gridster-item>
  <custom-directive></custom-directive>
</div>
```
but for it to work the contents should be like this:
```html
<div gridster-item>
  <custom-directive>
    <div class=".the-handle">dragging handle</div>
  </custom-directive>
</div>
```

When I put the template directly in the directive definition using the template property I don't notice this issue. Not sure if that's because something different is actually happening or if it's because the race condition gets tightened up to the point that it's unlikely to see.

Not sure when I'll have time to look at this but I wanted to get it tracked here right away.

**Comment by mikehaas763** [comment]
I can really see how it's a race condition when I set breakpoints within the code and give whatever is happening time to finish, then the custom handle selector works as expected.

**Comment by danomatic** [comment]
Thanks for reporting this.  It's a somewhat known issue that has existed in previous versions.  It's unclear how to solve it.  The best solution would be to delegate the mouse-down event handling to the parent container and then check if the clicked element matches the selector, but without full jQuery we don't have functions like "is" or event delegation to make that possible.

**Comment by danomatic** [comment]
Angular has a .ready event handler that is called when the template has rendered.  Fixed in 9f140003ecde3a07aa8eaa88e28f7e08f5b5a357
